### PR TITLE
some cleanup and refactoring codes of analysis-nori & analysis-kuromoji

### PR DESCRIPTION
**Pull request by twosom**
### Description
Some cleanup and refactoring codes of `analysis-nori` and `analysis-kuromoji`
- tried to improving code readability.
- add `final` keyword in `KatakanaRomanizer#INSTANCE`

### CODE REVIEW

File: lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseBaseFormFilter.java [hunk 1]
```diff
@@ -44,16 +44,18 @@ public JapaneseBaseFormFilter(TokenStream input) {
...
+
+    if (keywordAtt.isKeyword()) {
+      return true;
+    }
+
+    String baseForm = basicFormAtt.getBaseForm();
+    if (baseForm != null) {
+      termAtt.setEmpty().append(baseForm);
+    }
+    return true;
```
Comment (rmuir) [inline lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseBaseFormFilter.java:59]: i don't understand the improvement here. it is clearer to be able to see the "decision tree" than to do multiple returns like this.

**Comment by rmuir** [comment]
i dont understand the benefit, personally I think it makes code more difficult to read.

File: lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseBaseFormFilter.java (hunk 1 above)
Comment (twosom) [reply lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseBaseFormFilter.java:59]: thanks.
I'll change back this code.
🫠

File: lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseCompletionFilter.java [hunk 2]
```diff
@@ -98,20 +98,16 @@ public void reset() throws IOException {
...
+    if (tokenGenerator.hasNext() == false) {
       return false;
     }
+
+    clearAttributes();
+    CompletionToken token = tokenGenerator.next();
+    termAttr.setEmpty().append(token.term);
+    posIncAtt.setPositionIncrement(token.isFirst ? 1 : 0);
+    offsetAtt.setOffset(token.startOffset, token.endOffset);
+    return true;
```
Comment (twosom) [inline lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/JapaneseCompletionFilter.java:110]: @rmuir 
In my opinion, this part is much readable for me.
can i ask your opinion?

**Pull request closed without merging**